snuba/datasets/storages/profiles.py

```diff
@@ -0,0 +1,88 @@
+from snuba.clickhouse.columns import UUID, ColumnSet, DateTime
+from snuba.clickhouse.columns import SchemaModifiers as Modifiers
+from snuba.clickhouse.columns import String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.profiles_processor import ProfilesMessageProcessor
+from snuba.datasets.schemas.tables import TableSchema, WritableTableSchema
+from snuba.datasets.storage import ReadableTableStorage, WritableTableStorage
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.table_storage import build_kafka_stream_loader_from_settings
+from snuba.query.processors.conditions_enforcer import OrgIdEnforcer, ProjectIdEnforcer
+from snuba.query.processors.table_rate_limit import TableRateLimit
+from snuba.query.processors.type_converters.uuid_column_processor import (
+    UUIDColumnProcessor,
+)
+from snuba.utils.streams.topics import Topic
+
+PROFILES_LOCAL_TABLE_NAME = "profiles_local"
+PROFILES_DIST_TABLE_NAME = "profiles_dist"
+
+
+processors = [
+    UUIDColumnProcessor(set(["transaction_id", "trace_id"])),
+    TableRateLimit(),
+]
+
+loader = build_kafka_stream_loader_from_settings(
+    processor=ProfilesMessageProcessor(), default_topic=Topic.PROFILES,
+)
+
+readable_columns = ColumnSet(
+    [
+        ("organization_id", UInt(64)),
+        ("project_id", UInt(64)),
+        ("transaction_id", UUID()),
+        ("received", DateTime()),
+        ("profile", String()),
+        ("symbols", String()),
+        ("android_api_level", UInt(32, Modifiers(nullable=True))),
+        ("device_classification", String()),
+        ("device_locale", String()),
+        ("device_manufacturer", String()),
+        ("device_model", String()),
+        ("device_os_build_number", String(Modifiers(nullable=True))),
+        ("device_os_name", String()),
+        ("device_os_version", String()),
+        ("duration_ns", UInt(64)),
+        ("environment", String(Modifiers(nullable=True))),
+        ("platform", String()),
+        ("trace_id", UUID()),
+        ("transaction_name", String()),
+        ("version_name", String()),
+        ("version_code", String()),
+    ]
+)
+
+readable_schema = TableSchema(
+    columns=readable_columns,
+    local_table_name=PROFILES_LOCAL_TABLE_NAME,
+    dist_table_name=PROFILES_DIST_TABLE_NAME,
+    storage_set_key=StorageSetKey.PROFILES,
+)
+
+readable_storage = ReadableTableStorage(
+    storage_key=StorageKey.PROFILES,
+    storage_set_key=StorageSetKey.PROFILES,
+    schema=readable_schema,
+    query_processors=processors,
+)
```

Comment:
You do not need these. 
You have one table only, which is read/write. 
`writable_storage` will do for both.
Actually you have to use only one. Otherwise the storage key is assigned to two storages.